common/src/main/java/org/opensearch/ml/common/utils/StringUtils.java

```diff
@@ -60,6 +63,12 @@ public class StringUtils {
         + "      return input;"
         + "\n    }\n";
 
+    // Regex allows letters, digits, spaces, hyphens, underscores, and dots.
+    private static final Pattern SAFE_INPUT_PATTERN = Pattern.compile("^[\\p{L}\\p{N}\\s.,!?():@\\-_'\"]*$");
```

Comment:
We might not want to have the regex match empty string, right? So use `+` instead of `*`?

Reply:
That's a good point. So for name I understand we don't want empty string, but description should be optional. Which is why I kept `+` what do you think?

Reply:
Thanks for the comment. I refactored my PR in a way, so that it can be used for required or non required fields both. Please check.

Reply:
Thanks for the addressing, LGTM.